pkgs/by-name/co/codecrafters-cli/package.nix

```
@@ -0,0 +1,53 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  git,
+}:
+
+buildGoModule rec {
+  pname = "codecrafters-cli";
+  version = "34";
+
+  src = fetchFromGitHub {
+    owner = "codecrafters-io";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-+daBduhjUt7lrjTdXgm1qpZN1oEHC3Nx+ZiOLoO5+s4=";
+    # A shortened git commit hash is part of the version output, and is
+    # needed at build time. Use the `.git` directory to retrieve the
+    # commit SHA, and remove the directory afterwards since it is not needed
+    # after that.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD | cut --characters=1-7 > $out/COMMIT
+      find "$out" -name .git -print0 | xargs -0 rm -rf
```

Retrieving the `COMMIT` can be shortened as follows:
```suggestion
      git -C $"$out" rev-parse --short=7 HEAD > $out/COMMIT
      find "$out" -name .git -print0 | xargs -0 rm -rf
```

Out of curiosity: What is the motivation for `find … | xargs -0 rm…`, instead of simply `rm -rf "$out/.git"`?

---

Good Question, it was a suggested codechange by @tomodachi94

---

It's what's been done in previous packages (see https://github.com/NixOS/nixpkgs/pull/232557#discussion_r1271327627). I don't have strong feelings about keeping this code or replacing it with `rm -rf`.

---

After looking a bit, I've seen both being used. So ig we can just leave it there, since it works

---

Feel free to keep as is, yet I think `rm` is likely (much) more performant and I'm genuinely curious for what reasons the `find… | xargs` approach is used, tbh I'm a bit stumped 🙂